src/profile/hooks/useProfileLoadTracker.ts

```diff
@@ -0,0 +1,139 @@
+import { useCallback, useContext, useEffect, useRef } from 'react';
+
+import { ProfileContext, ProfileContextData } from '../context/ProfileContext';
+import parseGraphQLError from '../helpers/parseGraphQLError';
+import { QueryResult } from './useProfileQuery';
+
+const STATUS_FAILED = -1;
+const STATUS_WAITING = 0;
+const STATUS_SHOULD_LOAD = 1;
+const STATUS_LOADING = 2;
+const STATUS_SUCCESS = 3;
+
+export type Status =
+  | typeof STATUS_WAITING
+  | typeof STATUS_LOADING
+  | typeof STATUS_FAILED
+  | typeof STATUS_SUCCESS
+  | typeof STATUS_SHOULD_LOAD;
+
+export type useProfileLoaderHookReturnType = {
+  isProfileLoadComplete: () => boolean;
+  hasExistingProfile: () => boolean;
+  didProfileLoadFail: () => boolean;
+  reloadProfile: () => Promise<QueryResult>;
+  getStatus: () => Status;
+  getProfile: () => ReturnType<ProfileContextData['getProfile']>;
+};
+
+export function shouldLoad(currentStatus?: Status): boolean {
+  return currentStatus === STATUS_SHOULD_LOAD;
+}
+
+export function getShouldLoadStatus(): Status {
+  return STATUS_SHOULD_LOAD;
+}
+
+export function isLoading(currentStatus?: Status): boolean {
+  return currentStatus === STATUS_LOADING;
+}
+
+export function getLoadingStatus(): Status {
+  return STATUS_LOADING;
+}
+
+export function didLoadFail(currentStatus?: Status): boolean {
+  return currentStatus === STATUS_FAILED;
+}
+
+export function getLoadFailedStatus(): Status {
+  return STATUS_FAILED;
+}
```

Comment:
It doesn't seem too useful to have this kind of small helpers that just basically give another name for a simple statement. Even if the new name is useful as an internal detail, I don't see why these should be exported.

Reply:
I agree. But they have to be exported for tests. Other option is to export all STATUS_xxx constants, but I think this is cleaner.

Reply:
I would prefer exporting the `STATUS_*` constants. They kind of form an enum and are thus grouped to a whole, whereas a bunch of functions are just separate functions without that much connection to each other. But JS/TS is weird to me anyway, so perhaps instead of a neat enum it's customary to create a bunch of oneliner functions :shrug:

Reply:
Changed to use STATUS_*